apps/service_providers/tests/test_messaging_providers.py

```diff
@@ -0,0 +1,70 @@
+import mock
+import pytest
+from pydantic import ValidationError
+
+from apps.channels.models import ChannelPlatform
+from apps.service_providers.models import MessagingProvider, MessagingProviderType
+
+
+def test_twilio_messaging_provider(team):
+    _test_messaging_provider(
+        team,
+        MessagingProviderType.twilio,
+        data={
+            "auth_token": "test_token",
+            "account_sid": "account_sid",
+        },
+    )
+
+
+@pytest.mark.parametrize(
+    "config_key",
+    [
+        "auth_token",
+        "account_sid",
+    ],
+)
+def test_twilio_messaging_provider_error(config_key):
+    """Test that any missing param causes failure"""
+    form = MessagingProviderType.twilio.form_cls(
+        data={
+            "auth_token": "test_key",
+            "account_sid": "test_secret",
+        }
+    )
+    assert form.is_valid()
+    form.cleaned_data.pop(config_key)
+    _test_messaging_provider_error(MessagingProviderType.twilio, data=form.cleaned_data)
+
+
+@pytest.mark.parametrize(
+    "platform,expected_provider_types",
+    [
+        ("whatsapp", ["twilio"]),
+        ("telegram", []),
```

Comment:
Is the plan to add a provider for telegram later?

Reply:
Oh, no. This was just to test that no providers should be returned for platforms that doesn't have any providers